concourse/steps/release.py

```diff
@@ -23,19 +28,303 @@
 )
 
 
-def rebase(
-    git_helper,
-    upstream_ref,
-):
-    upstream_commit_sha = git_helper.fetch_head(upstream_ref).hexsha
-    git_helper.rebase(commit_ish=upstream_commit_sha)
+class ReleaseContext(object):
+    def set_release_commit_sha(self, release_commit_sha):
+        self._release_commit_sha = release_commit_sha
+
+    def release_commit_sha(self):
+        return self._release_commit_sha
+
+
+class TransactionalStep(object):
+    ''' Provides a framework to implement some action in an atomic fashion.
+
+    Instances of this class are intended to perform exactly one change in their apply-method
+    and provide the corresponding undo-functionality in the revert-method
+    '''
+    def __init__(self, release_context: ReleaseContext):
```

Comment:
so. as I elaborated above. the release-ctx should be named (and crafted) more generically. either by changing as described before. or by (this would be the easies approach) changing it to a generic object, such as a dict or a SimpleNamespace.